frontend/src/views/SettingWorkspaceGeneral.vue

```diff
@@ -1,56 +1,175 @@
+          </button>
+        </div>
+      </div>
     </div>
   </div>
+  <FeatureModal
+    v-if="state.showFeatureModal"
+    feature="bb.feature.branding"
+    @cancel="state.showFeatureModal = false"
+  />
 </template>
 
-<script lang="ts">
-import { computed, defineComponent, reactive } from "vue";
+<script lang="ts" setup>
+import { computed, reactive } from "vue";
 import { useStore } from "vuex";
 import { isOwner } from "../utils";
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-interface LocalState {}
+interface LocalState {
+  displayName?: string;
+  logoUrl?: string;
+  logoFile: File | null;
+  loading: boolean;
+  showFeatureModal: boolean;
+}
 
-export default defineComponent({
-  name: "SettingWorkspaceGeneral",
-  setup() {
-    const store = useStore();
+const maxFileSizeInMiB = 2;
 
-    const state = reactive<LocalState>({});
+// getImageUrlFromBase64String will convert the image base64 string to the HTML image url.
+const getImageUrlFromBase64String = (
+  base64String: string | undefined
+): string => {
+  if (!base64String) return "";
 
-    const currentUser = computed(() => store.getters["auth/currentUser"]());
+  const decodedData = window.atob(base64String);
+  const uInt8Array = new Uint8Array(decodedData.length);
 
-    const allowEdit = computed((): boolean => {
-      return isOwner(currentUser.value.role);
-    });
+  // Insert all character code into uInt8Array
+  for (let i = 0; i < decodedData.length; ++i) {
+    uInt8Array[i] = decodedData.charCodeAt(i);
+  }
 
-    const allowSave = computed((): boolean => {
-      return false;
-    });
+  const blob = new Blob([uInt8Array]);
+  return URL.createObjectURL(blob);
+};
 
-    const doSave = () => {
-      // do nothing
-    };
+const store = useStore();
 
-    return {
-      state,
-      allowEdit,
-      allowSave,
-      doSave,
-    };
-  },
-  data() {
-    return { placeholder: "{{ DB_NAME_PLACEHOLDER }}" };
-  },
+const state = reactive<LocalState>({
+  displayName: "",
+  logoUrl: "",
+  logoFile: null,
+  loading: false,
+  showFeatureModal: false,
 });
+
+const currentUser = computed(() => store.getters["auth/currentUser"]());
+
+const allowEdit = computed((): boolean => {
+  return isOwner(currentUser.value.role);
+});
+
+const valid = computed((): boolean => {
+  return !!state.displayName || !!state.logoFile
+});
+
+const changed = computed((): boolean => {
+  return false;
+});
+
+const allowSave = computed((): boolean => {
+  return allowEdit.value && changed.value && valid.value;
+});
+
+const hasBrandingFeature = computed((): boolean => {
+  return store.getters["subscription/feature"]("bb.feature.branding");
+});
+
+const uploadLogo = () => {
+  if (!hasBrandingFeature.value) {
+    state.showFeatureModal = true;
+    return;
+  }
+  // TODO: upload logo
+};
+
+const onLogoSelect = (file: File) => {
+  state.logoFile = file;
+  state.logoUrl = URL.createObjectURL(file);
+};
+
+const supportImageExtensions = [
```

Comment:
nit: I would put const list to the top of the script section

Reply:
I've optimized and other feature updates in the following PR, but blocked by this one.